src/librbd/internal.cc

```diff
@@ -1575,13 +1605,6 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
         enable_flags |= RBD_FLAG_FAST_DIFF_INVALID;
         features_mask |= (RBD_FEATURE_OBJECT_MAP | RBD_FEATURE_EXCLUSIVE_LOCK);
       }
-
-      if (enable_flags != 0) {
-        r = update_all_flags(ictx, enable_flags, enable_flags);
```

Comment:
I wouldn't move this down -- we want to flag the object map as invalid before anyone gets a chance to use it.